File: source/aoBlogs2/Models/GroupModel.vb

Imports Contensive.BaseClasses
Imports Contensive.Models.Db

Namespace Models
    Public Class GroupModel
        Inherits Contensive.Models.Db.PersonModel
        '
        '====================================================================================================
        '
        Public Shared Function GetBlockingGroups(cp As CPBaseClass, BlogCategoryID As Integer) As List(Of GroupModel)
            '
            Dim result As New List(Of GroupModel)
            '           
            For Each BlogCategoryGroupRule In DbBaseModel.createList(Of BlogCategoryGroupRulesModel)(cp, "BlogCategoryID=" & BlogCategoryID)
                result.Add(DbBaseModel.create(Of GroupModel)(cp, BlogCategoryGroupRule.GroupID))
            Next
            Return result
        End Function


    End Class
End Namespace

File: source/aoBlogs2/Models/Db/GroupModel.cs

using System.Collections.Generic;
using Contensive.BaseClasses;

namespace Contensive.Blog.Models {
    public class GroupModel : Contensive.Models.Db.GroupModel {
        //        
        public static List<GroupModel> GetBlockingGroups(CPBaseClass cp, int BlogCategoryID) {
            // 
            var result = new List<GroupModel>();
            // 
            foreach (var BlogCategoryGroupRule in createList<BlogCategoryGroupRulesModel>(cp, "BlogCategoryID=" + BlogCategoryID))
                result.Add(create<GroupModel>(cp, BlogCategoryGroupRule.GroupID));
            return result;
        }

    }
}